pkgs/development/compilers/ghc/head.nix

```diff
@@ -15,16 +15,26 @@
 
 , # If enabled, GHC will be built with the GPL-free but slower integer-simple
   # library instead of the faster but GPLed integer-gmp library.
-  enableIntegerSimple ? false, gmp ? null
+  enableIntegerSimple ? targetPlatform != hostPlatform, gmp ? null
 
 , # If enabled, use -fPIC when compiling static libs.
   enableRelocatedStaticLibs ? targetPlatform != hostPlatform
 
 , # Whether to build dynamic libs for the standard library (on the target
   # platform). Static libs are always built.
-  enableShared ? true
+  #
+  # We don't do this by default if target != host, or if we target windows.
+  enableShared ? (targetPlatform == hostPlatform && !targetPlatform.isWindows)
+
+, # Whetherto build terminfo.
+  enableTerminfo ? !targetPlatform.isWindows
```

Comment:
Does terminfo/ncurses not work on windows?

If it was failing to find ncurses headers/libs see the fix on my musl PR (remove attempt to specify where to find them).

Reply:
In the case of cross compiling to windows, we don't want terminfo, the GHC build system disables terminfoin that case anyway.

Reply:
It only disables it in some cross build "flavour"s, which we don't have to use as-is.  Works for me building cross-aarch64-musl and such.  Anyway sounds like you're saying windows in particular shouldn't use this, which I'll take your word for.